### PR TITLE
ofAVFoundationPlayer release used resources on dealloc

### PR DESCRIPTION
**Pull request by armadillu**
#4414

### CODE REVIEW

File: libs/openFrameworks/video/ofAVFoundationVideoPlayer.m [hunk 1]
```diff
@@ -122,19 +122,33 @@ - (void)dealloc
...
+	if(self.assetReader != nil) {
+		[self.assetReader cancelReading];
+		self.assetReader = nil;
+		self.assetReaderVideoTrackOutput = nil;
+		self.assetReaderAudioTrackOutput = nil;
+	}
+
+	if(self.asset != nil) {
+		self.asset = nil;
+	}
```
Comment (ofTheo) [inline libs/openFrameworks/video/ofAVFoundationVideoPlayer.m]: is this really releasing the asset? - 
I think setting it to nil is not releasing it. 

ping @julapy @danoli3  @admsyn

**Comment by 2bbb** [comment]
i think, if ARC is enabled then this code will release assets.
but it can call `[super dealloc];` at the last.
so, ARC is not enabled on this code.
maybe we must do call release for each objs.

PS.
why do you check nil?
i think this can be more simple like:

```
self.assetReaderVideoTrackOutput = nil;
[self.assetReader cancelReading];
self.assetReader = nil;
self.assetReaderVideoTrackOutput = nil;
self.assetReaderAudioTrackOutput = nil;
self.asset = nil;
```

**Comment by armadillu** [comment]
I left those in bc I was checking for Release Counts in my debug code, but yeah we can drop most of these; not the one with self.assetReader though.

@ofTheo with ARC it will get auto-released. I really don't like the pattern and I prefer the usual alloc - release strategy, but those object haven't been allocated at creation, so they don't need a release. The release pool will release them if their RefCount is 0

**Comment by ofTheo** [comment]
ah okay - I was a little confused as I didn't realize setting a strong reference to nil decreased the refcount and essentially releases the reference.  

so this is a ton of stuff which isn't getting released!
crazy. going to merge.

**Comment by i-n-g-o** [comment]
`assetReader, assetReaderVideoTrackOutput, assetReaderAudioTrackOutput` and `asset` are defined as properties retaining the object. that makes the object being release when they are set to `nil`.
setting them to nil should be enough.

**Pull request merged**